src/credentials/examples/DefaultDeviceAttestationVerifier.cpp

```diff
@@ -196,118 +195,144 @@ class DefaultDACVerifier : public DeviceAttestationVerifier
-                                                                               const ByteSpan & dacDerBuffer,
-                                                                               const ByteSpan & attestationNonce)
+CHIP_ERROR DefaultDACVerifier::VerifyAttestationInformation(const ByteSpan & attestationInfoBuffer,
+                                                            const ByteSpan & attestationChallengeBuffer,
+                                                            const ByteSpan & attestationSignatureBuffer,
+                                                            const ByteSpan & paiDerBuffer, const ByteSpan & dacDerBuffer,
+                                                            const ByteSpan & attestationNonce,
+                                                            Callback::Callback<OnAttestationInformationVerification> * onCompletion)
 {
-    VerifyOrReturnError(!attestationInfoBuffer.empty() && !attestationChallengeBuffer.empty() &&
-                            !attestationSignatureBuffer.empty() && !paiDerBuffer.empty() && !dacDerBuffer.empty() &&
-                            !attestationNonce.empty(),
-                        AttestationVerificationResult::kInvalidArgument);
+    CHIP_ERROR error                               = CHIP_NO_ERROR;
+    AttestationVerificationResult attestationError = AttestationVerificationResult::kSuccess;
 
     VendorId dacVendorId = VendorId::NotSpecified;
+    Platform::ScopedMemoryBuffer<uint8_t> paaCert;
+    MutableByteSpan paaDerBuffer;
+
+    VerifyOrExit(!attestationInfoBuffer.empty() && !attestationChallengeBuffer.empty() && !attestationSignatureBuffer.empty() &&
+                     !paiDerBuffer.empty() && !dacDerBuffer.empty() && !attestationNonce.empty() && onCompletion != nullptr,
+                 (attestationError = AttestationVerificationResult::kInvalidArgument, error = CHIP_ERROR_INVALID_ARGUMENT));
+
     // match DAC and PAI VIDs
     {
         uint16_t paiVid = VendorId::NotSpecified;
         uint16_t dacVid = VendorId::NotSpecified;
 
-        VerifyOrReturnError(ExtractDNAttributeFromX509Cert(MatterOid::kVendorId, paiDerBuffer, paiVid) == CHIP_NO_ERROR,
-                            AttestationVerificationResult::kPaiFormatInvalid);
-        VerifyOrReturnError(ExtractDNAttributeFromX509Cert(MatterOid::kVendorId, dacDerBuffer, dacVid) == CHIP_NO_ERROR,
-                            AttestationVerificationResult::kDacFormatInvalid);
+        VerifyOrExit((error = ExtractDNAttributeFromX509Cert(MatterOid::kVendorId, paiDerBuffer, paiVid)) == CHIP_NO_ERROR,
+                     attestationError = AttestationVerificationResult::kPaiFormatInvalid);
+        VerifyOrExit((error = ExtractDNAttributeFromX509Cert(MatterOid::kVendorId, dacDerBuffer, dacVid)) == CHIP_NO_ERROR,
+                     attestationError = AttestationVerificationResult::kDacFormatInvalid);
```

Comment:
These are dual-side-effect macros. Very hard to understand. Please only use `attestationError` value to map to CHIP_NO_ERROR/CHIP_ERROR...

Since the result is async, don't expect to return errors to caller, only expect callback to have been called

Reply:
Addressed in 1a5487c